src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,216 @@
+package org.openkilda.functionaltests.spec.links
+
+import static groovyx.gpars.GParsPool.withPool
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.functionaltests.extension.tags.Tag.HARDWARE
+import static org.openkilda.testing.Constants.RULES_DELETION_TIME
+import static org.openkilda.testing.Constants.RULES_INSTALLATION_TIME
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.command.switches.DeleteRulesAction
+import org.openkilda.messaging.command.switches.InstallRulesAction
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.SwitchChangeType
+import org.openkilda.model.IslStatus
+import org.openkilda.model.SwitchFeature
+import org.openkilda.testing.model.topology.TopologyDefinition.Isl
+
+import spock.lang.See
+
+@Tags(HARDWARE)
+// virtual env doesn't support round trip latency
+@See("https://github.com/telstra/open-kilda/tree/develop/docs/design/network-discovery")
+class RoundTripIslSpec extends HealthCheckSpecification {
+
+    /*we need this variable because it takes more time to DEACTIVATE a switch
+    via the 'knockoutSwitch' method on the stage env*/
+    Integer customWaitOffset = WAIT_OFFSET * 4
+
+    @Tidy
+    def "A round trip latency ISL doesn't go down when one switch lose connection to FL"() {
+        given: "A switch with/without round trip latency ISLs"
+        def roundTripIsls
+        def nonRoundTripIsls
+        def swToDeactivate = topology.activeSwitches.find { sw ->
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
+                roundTripIsls = topology.getRelatedIsls(sw).findAll {
+                    it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                nonRoundTripIsls = topology.getRelatedIsls(sw).findAll {
+                    !it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                roundTripIsls && nonRoundTripIsls
+            }
+        } ?: assumeTrue("Wasn't able to find a switch with suitable links", false)
+
+        when: "Simulate connection lose between the switch and FL, the switch becomes DEACTIVATED and remains operable"
+        lockKeeper.knockoutSwitch(swToDeactivate)
+        def isSwDeactivated = true
+        Wrappers.wait(customWaitOffset) {
+            assert northbound.getSwitch(swToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        and: "Wait discoveryTimeout"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
```

Comment:
still don't understand the ` + WAIT_OFFSET / 2` thing. why not sleep for discoveryTimeout and then use _wait_ in order to wait for FAILED status.?

Reply:
I thought it would be better to do it in the same line and then do the verifications.
I am fine with your proposal. This line is refactored